### DISCUSSION TITLE
Tech and Block inspection: Properties to be implemented

### DISCUSSION THREAD

**Discussion by AcKoucher** [The-OpenROAD-Project/OpenROAD#3477] [General]
I am opening this discussion to record some properties that were not included in the dbTech and dbBlock inspection implemented in #3458 and to make it easier to make future questions related to these properties.

As @maliberty did not want to overload the PR, I left out some of the properties that @gadfort asked to include in the inspection. The ones I did not implement are the ones available through the following methods:

dbBlock:
`getTopModule()`
`getChildren()`
`getVias()`

dbTech:
`getViaRules()`

**Comment by AcKoucher** [comment]
@maliberty I am working in the remaining properties and I have the following doubt:

According to db.h, A TechViaLayerRule is the element that represents a LEF VIARULE LAYER.

The type dbTechViaLayerRule has methods such as:
`bool hasEnclosure();`
`void getEnclosure(int& overhang1, int& overhang2);`
`bool hasOverhang();`
`int getOverhang();`
`...`


But in the LEF/DEF reference the VIARULE LAYER can have only a WIDTH range and a DIRECTION. Dummy example:
```
VIARULE Via1Array-0
  LAYER via1 ;
    DIRECTION HORIZONTAL ;
    WIDTH 0.035 TO 0.035 ;
  LAYER via2 ;
    DIRECTION VERTICAL ;
    WIDTH 0.035 TO 0.035 ;
  VIA via1_0 ;
  VIA via1_1 ;
END Via1Array-0
```

So are these other methods correct? When inspecting a TechViaLayerRule just the WIDTH and DIRECTION should appear?

**Reply by maliberty** [reply]
Apparently it is also used for VIARULE GENERATE statements as well.  Those can have an ENCLOSURE statement.

**Reply by maliberty** [reply]
See https://github.com/The-OpenROAD-Project/OpenROAD/blob/3b67f4e01529f6add7eccb20e952fa1140ff972e/src/odb/src/lefin/lefin.cpp#L2043
